src/evict/evict_page.c

```diff
@@ -675,63 +677,75 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
-    /* Acquire a snapshot if coming through eviction thread route. */
+    /*
+     * Acquire a snapshot if coming through the eviction thread route. Also, if we have entered
+     * eviction through application threads and we have a transaction snapshot, we will use our
+     * existing snapshot to evict pages that are not globally visible based on the last_running
+     * transaction. Avoid using snapshots when application transactions are in the final stages of
+     * commit or rollback as they have already released the snapshot. Otherwise, it becomes harder
+     * in the later part of the code to detect updates the belonged to the last running application
+     * transaction.
+     */
 
     /*
      * TODO: We are deliberately not using a snapshot when checkpoint is active. This will ensure
      * that point-in-time checkpoints have a consistent version of data. Remove this condition once
-     * fuzzy transaction ID based checkpoints work is merged.
+     * fuzzy transaction ID based checkpoints work is merged (WT-6673).
      */
-    if (FLD_ISSET(evict_flags, WT_REC_EVICTION_THREAD) && !WT_IS_HS(S2BT(session)) &&
-      !conn->txn_global.checkpoint_running) {
-        /*
-         * We rely on the fact that the eviction threads are created with read committed isolation
-         * by default. If this fact doesn't hold anymore in future, we have to force isolation
-         * level.
-         */
-        WT_ASSERT(session, session->txn->isolation == WT_ISO_READ_COMMITTED);
+    use_snapshot_for_app_thread = !F_ISSET(session, WT_SESSION_INTERNAL) &&
+      !WT_IS_METADATA(session->dhandle) && WT_SESSION_TXN_SHARED(session)->id != WT_TXN_NONE &&
+      F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT);
+    use_snapshot_for_eviction_thread = FLD_ISSET(evict_flags, WT_REC_EVICTION_THREAD);
```

Comment:
The simplicity of that boolean makes me think it really isn't necessary. I'd still prefer only one. If you really want to keep it, can it be called `is_eviction_thread` instead? That more accurately describes what it is.